### DISCUSSION TITLE
astro unadd <integration>

### DISCUSSION THREAD

**Discussion by schneedotdev** [withastro/roadmap#522] [Proposal]
# Summary

Unadd a previous `astro add <integration>` with a simple `unadd` command.

## Should 
Remove added files and changes to `package.json`, `pacakge-lock.json`, `astro.config.mjs`, and other affected areas of a project. 

## Instead of
Going through your project and manually uninstalling and removing the additions.

# Background & Motivation

After making frequent integration changes, I would love a seamless CLI command to do the unadd work for me just as the `add` command does so flawlessly.

_It should be just as simple to unadd changes as it is to add them._

# Goals

- Add a command that undoes changes made to a project with `astro add <integration>`.

# Example

After running `astro add <integration>` and approving the project changes, I would want to be able to run the command `astro unadd <integration>` at a later time that removes all of the changes that were made to my project (given I no longer wanted the integration for my project).

**Comment by ematipico** [comment]
I want to propose the `remove` name instead of `unadd` . "unadd" doesn't feel right. 

Removing a dependency will take more work than adding an integration (from the development point of view).

**Reply by schneedotdev** [reply]
I like `remove`, but it is important to note that the `add` command does more than add a dependency. As such, the `unadd` command would encompass the same area of behavior as `add`. The context focuses on integration, which considers dependencies and code additions.

That said, I see no problem and think `remove` is a great suggestion. I'm happy for whatever makes it easiest for users to understand.

**Reply by T3sT3ro** [reply]
A common name for that case is `purge`.

**Reply by dei8bit** [reply]
i think is more intuitive **unadd**,  when you remember one remember the other; instead purge , remove, delete, etc.. are different words

**Comment by louiss0** [comment]
I would also like this. This feature would make life so easy. I think it should either ask to turn of the server or just turn it of. If you are using an integration that injects scripts then you would have to turn off the server and fix the errors in the code anyway.

**Comment by ghost** [comment]
4.0 please?

**Comment by natemoo-re** [comment]
I don't know that we'll be able to prioritize this over other features since removing an integration is not very common, but contributions are welcome!

**Comment by T3sT3ro** [comment]
If this were to be added:

1. How should configuration files be handled? For example `tailwind.config.js`, I can see 3 possible routes:
  - log info/warn to console about remaining, possibly redundant files
  - delete them (ok if someone uses version control; bad if someone doesn't, e.g. novices)
  - log + interactive prompt for each removal
2. What about redundant configuration blocks in other files (e.g. hooks in astro config)?

Personally I would go for the third option with interactive removal.

**Comment by jsanchezpc** [comment]
what about `npm uninstall vue`???

**Comment by dei8bit** [comment]
So, _to date_, there is no way to remove something installed with **astro add**?  
- Should I manually search for it in my node modules folder and delete it? 
- Or is astro supposed to eradicate it during construction since it is not being used?

I ask because I really want to find the right way to do it and maintain my project well.

**Reply by delucis** [reply]
The right way to do this is with your package manager’s remove command.

For example, with `npm`:

```sh
npm uninstall PACKAGE-NAME
```

You’ll also need to double check `astro.config.mjs` to make sure the added package is not still being used there.

On the whole you can think of `astro add` as doing the following:

1. Installing a dependency (sometimes more than one)
2. Making some small changes to the Astro config to add the dependency there

So removing an integration involves doing the inverse of those steps:

1. Removing any use of the integration (most often in `astro.config.mjs`)
2. Uninstalling the dependencies

**Reply by dei8bit** [reply]
I had not read this...
The most usefull answer, really, really thanks  ♥
